transport-native-unix-common/src/main/java/io/netty/channel/unix/Socket.java

```diff
@@ -406,6 +406,8 @@ public final void setTrafficClass(int trafficClass) throws IOException {
         setTrafficClass(fd, trafficClass);
     }
 
+    public static native boolean isIPv6Available();
```

Comment:
nit: we could cache this if we want.

Reply:
+1 lets not expose the native method in the API. It also clarifies the semantics about how it will behave if we cache it coupled with `initialize()`. Also IPv6 maybe available, but the user just prefers to use IPv4 ... maybe `isIPv6Preferred` would be more representative of what this method provides?